Add NamedData to flat_tensor schema

lucylq opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #9125
* #9124
* __->__ #9123

This allows opaque data blobs inside PTD files.

Differential Revision: [D70530888](https://our.internmc.facebook.com/intern/diff/D70530888/)

lucylq committed:
Add NamedData to flat_tensor schema

This allows opaque data blobs inside PTD files.

Differential Revision: [D70530888](https://our.internmc.facebook.com/intern/diff/D70530888/)

[ghstack-poisoned]

swolchok commented:
code looks fine to me but I'm probably not the right person to review if schema changes are the right thing. let me know if there's nobody else ramped up either and I'll try to get context.

lucylq committed:
Update on "Add NamedData to flat_tensor schema"


This allows opaque data blobs inside PTD files.

Differential Revision: [D70530888](https://our.internmc.facebook.com/intern/diff/D70530888/)

[ghstack-poisoned]
